### DISCUSSION TITLE
Settings for JKBMS to recalculate SOC

### DISCUSSION THREAD

**Discussion by fast-quality** [Louisvdw/dbus-serialbattery#829] [Q&A]
Hello comunity,

I want to ask who I can solve the Problem with the SOC calibration on JK.

When I bring the batterie to max/min voltage limit, limits in JK configuration,
the SOC should be correct for a while.

But day by day, or after each charge/discharge cycle the SOC goes down by about 4-5%.
I will say It doesn´t reach to full anymore.
So I must bring it to the JK limit to get the 100% SOC back.
This i do once a week.

I want to ask the comunity, who you fix it.
I think I am not the only one with this problem 

With the new driver, there is a feature for it, but what are good settings to get it worked for the driver and for the JK.

Greeting from Germany
Martin

**Comment by mr-manuel** [comment]
What is your current config from the driver and JKBMS?

**Comment by fast-quality** [comment]
Hello Manuel,

Nice that you ask me, i have made some screenshots from the settings. The Problem annoys me for a long time and I have no good solution for it. I forced the JK to full SOC, by using the DVCC maximum charge voltage. When the batterie is full, I switch it to a higher level manually for a short time until I get the over voltage warning. The driver is the actual one from github, settings below. I installed it, for a few days and have not done all settings, because I had not many time, I think it may be possible to use the CVCM for the Problem, but what are good settings?

Before I used the old one 0.14* for a couple of months. The problem with the JK was the same. The driver worked fine. Another idea was that I connected a relay to one cell measurement and switch for a short a time another voltage source to the jk which forced it to set the SOC to 100%, But this all is not a good solution. In my opinion the JK must have a second separate parameter Set only for the SOC voltage levels. I wrote them 2 times already, but I get no answer. Firmware updates don´t seem possible either.

Greetings from Germany 
Martin

**Reply by mr-manuel** [reply]
Unfortunately there are no screenshots of the JKBMS settings and the `config.ini` is missing.

**Comment by fast-quality** [comment]
This is curious, because in my mailbox they are included. I will send the mail again

**Reply by mr-manuel** [reply]
This do not work via mail. You have to paste the images directly on GitHub.

**Reply by fast-quality** [reply]
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/af5f35eb-5cef-4414-92e8-b9d42419660a)

**Reply by fast-quality** [reply]
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/3ea20cc5-64f4-465e-87b4-a48a00398e91)

**Reply by fast-quality** [reply]
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/d131af11-94a9-479e-9d37-c1ff94bf8465)

**Reply by fast-quality** [reply]
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/277b1fd1-06bf-4892-9420-732ed0393e60)

**Comment by fast-quality** [comment]
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/7ca8d936-2e41-45fe-b2e7-19794546cde7)![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/45ad73c2-18d1-46b0-923e-b1d0958488bf)![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/5010b627-929c-4d4a-a7dd-eb327551d741)![image](https://github.com/Louisvdw/dbus-serialbattery/assets/120103744/7f54489d-c1d5-4892-8480-f284c0ec0045)

**Reply by mr-manuel** [reply]
`SOC_RESET_VOLTAGE` should be a bit under `OVP` setting from the JKBMS, but you have to try a bit on your system, since it depends also on the cell imbalance that you have.
* Try to increase (add) by `0.005` in steps, if the system does not switch to float mode, even if the target voltage `SOC_RESET_VOLTAGE * CELL_COUNT` is reached.
* Try to decrease (lower) by `0.005` in steps, if the system hits the `OVP` too fast, before all cells could be balanced and the system goes into protection mode multiple times.

You have `OVP` in the BMS settings on `3.575`, therefore I would reccomend this settings:

```ini
SOC_RESET_VOLTAGE  = 3.545
```

And set this to 7, you it tries to reset every 7 days as you like:

```ini
SOC_RESET_AFTER_DAYS = 7
```

I also extended the description of the config file, which will be merged in the next days: https://github.com/mr-manuel/venus-os_dbus-serialbattery/blob/631dfc96395a840c3b533ccb76d9d62b507966e3/etc/dbus-serialbattery/config.default.ini#L23-L45

**Comment by fast-quality** [comment]
Hello,

do anyone have an idea, what i can do? 
Is this problem normally ? Monday unit or something, maybe another bms works better ?
Can I easy set the CVCM settings, that the Charge Voltage run into the OVP.
I think 3.575 is a little bit high for this,.
Aren´t there anybody else with those problems.

Martin

**Reply by mr-manuel** [reply]
> I think 3.575 is a little bit high for this,.

It depends on your system. If you have a high cell imbalance everytime you try to reset to 100% SOC, then it could even be, that you have to set the OVP higher. If your cells are well balanced you also can try to set OVP to `3.500`. With the time the cells will be more imbalanced because of aging and you have to probably adjust the setting in a few months/years.

**Comment by fast-quality** [comment]
Hello Manuel,

thanks for your reply.
I have some questions and interesting observations regarding the issue

First is it always active or can it be disabled, maybe with the CVCM_Enable value ?
I think to set the SOC_RESET_VOLTAGE at 0.03 under the OVP from BMS seems to be a good approach,
please don´t hit me, but I think that it can be a little bit problematic, because the available energy from, example PV, is different from time to time. 
I mean, when there is enough power, it will work, because at least one cell voltage will be overdriven to OVP, due to a high enough charge current.
But when there is only a little bit power, the underrating (0.03) ensures that the OVP will not be reached, due to a to low charge current.

In past, I did it another way for a long time, what worked well.
I wait until the real Battery Voltage reached the set Charge voltage, the charge current is very low and the in Balance level is low. Then I switch the charge voltage to the higher level (manually, by using dvcc) and wait until the OVP is reached, then I turn the voltage limit back to standard. This works in both scenarios, by more power it goes faster, by little power it needs more time.

I think to automate it, this would be better. I haven´t tested your code yet, but I will do it promptly, I suspect that when OVP is reached and SOC is switched to equal or over 99% you set the charge voltage Back immediatley ?!

I like the functionality to active it by days (counter), as you have discribe it in the config.ini, because its easy.
Otherwise you could trigger it, when the SOC is below a certain level while hit the parameters: CVL is reached, Balance is ok, Current low or zero and delay timer.

I hope I give some interesting suggestions, I would like to code it myself, but I am to far away from those systems and have to less experience with it and a little bit of fear by making it wrong. My Home is Electro mechanical engineering which works with other coding languages.

Greetings Martin

**Reply by mr-manuel** [reply]
Just try it, you will see that it works well.

**Comment by fast-quality** [comment]
Hello Manuel,

i tried it, but it shows a strange behavior.

At first the CVL shows "bulk" and the voltage is at the setted parameter, all fine.
When the battery reached this voltage the SOC goes to 100%, fine too.
But after reached it, it still remain by "bulk" and the higt voltage ?!
I observed it for a while and the cvl is switching from "bulk" to "bulk (dynamic) + balancing (linear mode)"

I don´t understand why the CVL doesn´t switch back to max_cell_voltage ?
Is it necessary that the cells must be balanced at first, so that the driver do it ?

What I noticed is that in that under "IO" the Message "Allow to charge" is permanently toggling from "Yes" to "No" and back

Furthermore I have the Problem that the showed CCL not regulate the MPPT´s.
In detail, I have 2 Multis and 2 MPPT´s and 1 AC inverter. It seems to be, that the grid feed in regulation only works with the multi and extern inverter, but not for the MPPT´s. The voltage control works for it, but the current is everytime at max. I think when they give more power, as the battery needs, the multis should take over the excess, but they don´t do this.

I will wait for a while and will see what happened when the balancing is finished

greatings MArtin

**Reply by mr-manuel** [reply]
> Is it necessary that the cells must be balanced at first, so that the driver do it ?

Correct. And after all cells are balanced the system waits some more time `MAX_VOLTAGE_TIME_SEC`.

> What I noticed is that in that under "IO" the Message "Allow to charge" is permanently toggling from "Yes" to "No" and back

This because your OVP from the BMS is triggered. Try to lower `SOC_RESET_VOLTAGE` in `0.005` V steps.

> Furthermore I have the Problem that the showed CCL not regulate the MPPT´s.

Please open a question in the Victron Community for this. The driver cannot do much for this.

**Comment by fast-quality** [comment]
Hello Manuel,

I have tested the functionality and want to ask you if it is possible that you make a Point configurable.

I have the big Problem, that Victron doesn´t get the CCL working by using MPPT´s and DC feed-in.
When the driver switches the CVL to BULK_CELL_VOLTAGE the current goes very high (no limiting) and sometimes some Cell voltages goes to high.
It will be very cool when I can make a setting, that the BULK voltage switches back immediately when the SOC reached 99-100% without waiting for the voltage and balancing.

In my opinion the Balancing by MAX_CELL_VOLTAGE is alway enough. 
The Balancing by BULK_CELL_VOLTAGE stresses the batteries really, because the current limiting by Victron doesn´t work well.

Greetings Martin
